runtime/core/decoder/ov_asr_model.cc

```diff
@@ -0,0 +1,490 @@
+    if (it.real_time.count() > 0) {
+      totalTime += it.real_time;
+    }
+  }
+  stream << std::setw(20) << std::left
+         << "Total time: " + std::to_string(totalTime.count())
+         << " microseconds" << std::endl;
+  std::cout << std::endl;
+  std::cout << "Full device name: " << deviceName << std::endl;
+  std::cout << std::endl;
+  std::cout.flags(fmt);
+}
+
+// std::shared_ptr<ov::Core> OVAsrModel::core_ = std::make_shared<ov::Core>();
+OVAsrModel::~OVAsrModel() {}
+
+void OVAsrModel::InitEngineThreads(int core_number) {
+  core_ = std::make_shared<ov::Core>();
+  if (core_) {
+    core_->set_property("CPU", ov::num_streams(core_number));
+    core_->set_property("CPU", ov::affinity(ov::Affinity::NONE));
+    core_->set_property("CPU", ov::inference_num_threads(1));
+  } else {
+    std::cout << "OV core not init" << std::endl;
+  }
+}
+
+void OVAsrModel::Read(const std::string& model_dir) {
+  std::string encoder_ir_path = model_dir + "/encoder.xml";
+  std::string rescore_ir_path = model_dir + "/decoder.xml";
+  std::string ctc_ir_path = model_dir + "/ctc.xml";
+  auto ov_version = ov::get_openvino_version();
+  std::cout << "OPENVINO|VERSION|" << ov_version << std::endl;
+  try {
+    std::shared_ptr<ov::Model> encoder_model =
+        core_->read_model(encoder_ir_path);
+    std::map<std::string, ov::AnyMap> config;
+    config["CPU"] = {};
+    if (getenv("OPENVINO_PROFILE")) {
+      config["CPU"].emplace(ov::enable_profiling(true));
+    }
+    config["CPU"].emplace(
+        ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT));
+    // set some configurations if any
+    for (auto&& item : config) {
+      core_->set_property(item.first, item.second);
+    }
+    if (encoder_model) {
+      if (encoder_model->has_rt_info("framework")) {
+        auto metadata = encoder_model->get_rt_info<ov::AnyMap>("framework");
+
+        encoder_output_size_ = metadata["output_size"].as<int>();
+        num_blocks_ = metadata["num_blocks"].as<int>();
+        head_ = metadata["head"].as<int>();
+        cnn_module_kernel_ = metadata["cnn_module_kernel"].as<int>();
+        subsampling_rate_ = metadata["subsampling_rate"].as<int>();
+        right_context_ = metadata["right_context"].as<int>();
+        sos_ = metadata["sos_symbol"].as<int>();
+        eos_ = metadata["eos_symbol"].as<int>();
+        is_bidirectional_decoder_ =
+            metadata["is_bidirectional_decoder"].as<int>();
+        chunk_size_ = metadata["chunk_size"].as<int>();
+        num_left_chunks_ = metadata["left_chunks"].as<int>();
+      } else {
+        // in case that the model is exported without meta
+        encoder_output_size_ = 256;
+        num_blocks_ = 12;
+        head_ = 4;
+        cnn_module_kernel_ = 8;
+        subsampling_rate_ = 4;
+        right_context_ = 6;
+        sos_ = 4232;
+        eos_ = 4232;
+        is_bidirectional_decoder_ = 1;
+        chunk_size_ = 16;
+        num_left_chunks_ = -1;
+      }
```

Comment:
To inform users that metadata is missing and that we will fill in default values for xxx, it would be beneficial to log a message here.